code/game/machinery/doors/unpowered.dm

```diff
@@ -1,20 +1,43 @@
 /obj/machinery/door/unpowered
+	var/obj/item/lock_construct/Lock
+	var/lock_data
 
-/obj/machinery/door/unpowered/CollidedWith(atom/movable/AM)
-	if(src.locked)
-		return
-	..()
+/obj/machinery/door/unpowered/emag_act()
 	return
 
-
 /obj/machinery/door/unpowered/attackby(obj/item/I, mob/user, params)
-	if(locked)
+	if(istype(I, /obj/item/lock_construct/) && do_after(user, 10, target = src))
+		var/obj/item/lock_construct/L = I
+		if(!user.transferItemToLoc(I, src))
+			return
+		Lock = L
+		src.lock_data = L.lock_data
+		user.visible_message("[user] adds a lock to the door.")
+		desc = "[desc] has a lock engraved with a [L.lock_data]"
 		return
+	if(istype(I, /obj/item/key))
+		var/obj/item/key/K = I
+		if(Lock) //to check if we have added a lock, and should test the key
+			return Lock.check_key(K)
 	else
 		return ..()
 
-/obj/machinery/door/unpowered/emag_act()
-	return
+/obj/machinery/door/unpowered/CollidedWith(atom/movable/AM)
+	..()
+	if(Lock)
+		if(Lock.check_key())
+			return
+..()
```

Comment:
what does this do? it seems like a loose ..()

Reply:
This is mostly a placeholder for future changes

check_key() in the lock just ```return Lock.locked``` at the moment. In future I want things like traps/signallers to be attachable. So if the door is locked and you try to open it, it pings off the signaler or explodes the door, etc.